### ISSUE TITLE
Library lost binding with cookie-js 

### ISSUE THREAD

**Issue by raul-wiggot** [abersager/redux-persist-cookie-storage#11]
When my application loads I get this error:

Cannot read property 'get' of undefined
TypeError: Cannot read property 'get' of undefined
[...] /node_modules/redux-persist-cookie-storage/src/redux-persist-cookie-storage.js:35:27)

I located the error line and that part of the code checks if it's executed by a client or a server.

The title of the issue it's because I checked if **this.cookies** enters in one of the conditional blocks, which entered in the condition of the line 17, but as the error says, the variable gets undefined. Also I tried to hack it using the library of cookies directly, but I get the same error. I really don't know what its happening.

I'm using NextJS for server side rendering.

**Comment by ogheo** [comment]
I had the same issue and solved it in following way:

app.use(cookieParser());
...
const cookies = {cookies: req.cookies};
const { store } = configureStore(cookies);

**Comment by raul-wiggot** [comment]
@ogheo Do you have a complete example?

**Comment by spencewine** [comment]
I'm having this issue as well. @raul-wiggot did you every resolve this?

**Comment by raul-wiggot** [comment]
@spencewine I couldn't. Use another strategy to solve your problem :(

**Comment by dustinlarimer** [comment]
@raul-wiggot @spencewine Possibly related to https://github.com/abersager/redux-persist-cookie-storage/issues/10#issuecomment-373150153

>Additional confusion stems from the fact that the version shipped on npm is not what's currently in master but instead what's in the server-support branch.

Resolve this by installing [`cookies-js`](https://github.com/ScottHamper/Cookies) and passing it into the `CookieStorage()` constructor:

```ssh
$ npm install --save cookies-js
```

```js
import cookies from 'cookies-js';
import { CookieStorage } from 'redux-persist-cookie-storage';

const persistConfig = {
  key: 'root',
  storage: new CookieStorage(cookies)
};

const persistedReducer = persistReducer(persistConfig, rootReducer);
// https://github.com/rt2zz/redux-persist#basic-usage
```

**Comment by abersager** [comment]
@raul-wiggot Just published version 1.0.0. Please follow the instructions in the README and let me know if this works for you.
